From c752a2ce4c816a04846b3b4f8fba3edba0655ceb Mon Sep 17 00:00:00 2001
From: Wang Dongsheng <dongsheng.wang@nxp.com>
Date: Thu, 25 Feb 2016 10:43:19 +0800
Subject: [PATCH 564/752] ARM: LS1043: Add a wakeup source FTM0

Enable FTM0 for LS1043 platform.

Wang Dongsheng <dongsheng.wang@nxp.com>

[Original patch taken from QorIQ-SDK-V2.0-20160527-yocto]
Signed-off-by: Yanjiang Jin <yanjiang.jin@windriver.com>
---
 drivers/soc/fsl/Kconfig.arm                     | 16 ++--------------
 drivers/soc/fsl/Makefile                        |  1 +
 drivers/soc/fsl/{ls1 => layerscape}/Kconfig     |  3 +--
 drivers/soc/fsl/{ls1 => layerscape}/Makefile    |  0
 drivers/soc/fsl/{ls1 => layerscape}/ftm_alarm.c |  0
 5 files changed, 4 insertions(+), 16 deletions(-)
 rename drivers/soc/fsl/{ls1 => layerscape}/Kconfig (84%)
 rename drivers/soc/fsl/{ls1 => layerscape}/Makefile (100%)
 rename drivers/soc/fsl/{ls1 => layerscape}/ftm_alarm.c (100%)

diff --git a/drivers/soc/fsl/Kconfig.arm b/drivers/soc/fsl/Kconfig.arm
index 79aed5b..c22fb74 100644
--- a/drivers/soc/fsl/Kconfig.arm
+++ b/drivers/soc/fsl/Kconfig.arm
@@ -2,28 +2,16 @@
 # Freescale ARM SOC Drivers
 #
 
-config	LS1_SOC_DRIVERS
-	bool "LS1021A Soc Drivers"
-	depends on SOC_LS1021A
-	default n
-	help
-	Say y here to enable Freescale LS1021A Soc Device Drivers support.
-	The Soc Drivers provides the device driver that is a specific block
-	or feature on LS1021A platform.
-
 config LS_SOC_DRIVERS
 	bool "Layerscape Soc Drivers"
-	depends on ARCH_LAYERSCAPE
+	depends on ARCH_LAYERSCAPE || SOC_LS1021A
 	default n
 	help
 	Say y here to enable Freescale Layerscape Soc Device Drivers support.
 	The Soc Drivers provides the device driver that is a specific block
 	or feature on Layerscape platform.
 
-if LS1_SOC_DRIVERS
-	source "drivers/soc/fsl/ls1/Kconfig"
-endif
-
 if LS_SOC_DRIVERS
+	source "drivers/soc/fsl/layerscape/Kconfig"
 	source "drivers/soc/fsl/ls2-console/Kconfig"
 endif
diff --git a/drivers/soc/fsl/Makefile b/drivers/soc/fsl/Makefile
index 20923eb..7692ad9 100644
--- a/drivers/soc/fsl/Makefile
+++ b/drivers/soc/fsl/Makefile
@@ -5,4 +5,5 @@
 obj-$(CONFIG_QUICC_ENGINE)		+= qe/
 obj-$(CONFIG_CPM)			+= qe/
 obj-$(CONFIG_FSL_GUTS)			+= guts.o
+obj-$(CONFIG_LS_SOC_DRIVERS)		+= layerscape/
 obj-$(CONFIG_LS_SOC_DRIVERS)		+= ls2-console/
diff --git a/drivers/soc/fsl/ls1/Kconfig b/drivers/soc/fsl/layerscape/Kconfig
similarity index 84%
rename from drivers/soc/fsl/ls1/Kconfig
rename to drivers/soc/fsl/layerscape/Kconfig
index c9b04c4..e1373aa 100644
--- a/drivers/soc/fsl/ls1/Kconfig
+++ b/drivers/soc/fsl/layerscape/Kconfig
@@ -1,9 +1,8 @@
 #
-# LS-1 Soc drivers
+# Layerscape Soc drivers
 #
 config  FTM_ALARM
 	bool "FTM alarm driver"
-	depends on SOC_LS1021A
 	default n
 	help
 	Say y here to enable FTM alarm support.  The FTM alarm provides
diff --git a/drivers/soc/fsl/ls1/Makefile b/drivers/soc/fsl/layerscape/Makefile
similarity index 100%
rename from drivers/soc/fsl/ls1/Makefile
rename to drivers/soc/fsl/layerscape/Makefile
diff --git a/drivers/soc/fsl/ls1/ftm_alarm.c b/drivers/soc/fsl/layerscape/ftm_alarm.c
similarity index 100%
rename from drivers/soc/fsl/ls1/ftm_alarm.c
rename to drivers/soc/fsl/layerscape/ftm_alarm.c
-- 
2.9.3

